core/src/test/java/org/elasticsearch/search/sort/GeoDistanceSortBuilderTests.java

```diff
@@ -79,13 +81,13 @@ public static GeoDistanceSortBuilder randomGeoDistanceSortBuilder() {
             result.geoDistance(geoDistance(result.geoDistance()));
         }
         if (randomBoolean()) {
-            result.unit(unit(result.unit()));
+            result.unit(ESTestCase.randomValueOtherThan(result.unit(), () -> randomFrom(DistanceUnit.values())));
         }
         if (randomBoolean()) {
             result.order(RandomSortDataGenerator.order(null));
         }
         if (randomBoolean()) {
-            result.sortMode(mode(result.sortMode()));
+            result.sortMode(ESTestCase.randomValueOtherThan(SortMode.SUM, () -> randomFrom(SortMode.values())));
```

Comment:
Oh, I see in some of these cases you need to exclude something from the general enum like here. Nevermind my other comment.